src/ray/raylet/scheduling_resources.h

```diff
@@ -8,25 +8,64 @@
 
 #include "ray/raylet/format/node_manager_generated.h"
 
-#define EPSILON 0.00001
-
 namespace ray {
 
 namespace raylet {
 
 const std::string kCPU_ResourceLabel = "CPU";
 
-/// \brief Test if the quantity is within epsilon bounds of 0.
-///
-/// \param quantity: Quantity to check
-/// \return True if  -epsilon <= Quantity <= epsilon, False otherwise.
-bool EqualsZeroEpsilon(double quantity);
+/// \class FractionalResourceQuantity
+/// \brief Converts the resource quantities to an internal representation to
+/// avoid machine precision errors.
+class FractionalResourceQuantity {
+ public:
+  /// \brief Construct a FractionalResourceQuantity representing zero
+  /// resources.
+  FractionalResourceQuantity();
+
+  /// \brief Construct a FractionalResourceQuantity for implicit conversion of
+  /// operators.
+  FractionalResourceQuantity(int resource_quantity);
+
+  /// \brief Construct a FractionalResourceQuantity representing
+  /// resource_quantity.
+  FractionalResourceQuantity(std::string resource_name, double resource_quantity);
+
+  /// \brief Addition of FractionalResourceQuantity.
+  const FractionalResourceQuantity operator+(const FractionalResourceQuantity &rhs) const;
 
-/// \brief Test if the quantity is within epsilon bounds of 1.
-///
-/// \param quantity: Quantity to check
-/// \return True if  1 - epsilon <= Quantity <= 1 + epsilon, False otherwise.
-bool EqualsOneEpsilon(double quantity);
+  /// \brief Subtraction of FractionalResourceQuantity.
+  const FractionalResourceQuantity operator-(const FractionalResourceQuantity &rhs) const;
+
+  /// \brief Addition and assignment of FractionalResourceQuantity.
+  FractionalResourceQuantity &operator+=(const FractionalResourceQuantity &rhs);
+
+  /// \brief Subtraction and assignment of FractionalResourceQuantity.
+  FractionalResourceQuantity &operator-=(const FractionalResourceQuantity &rhs);
+
+  bool operator==(const FractionalResourceQuantity &rhs) const;
+  bool operator!=(const FractionalResourceQuantity &rhs) const;
+  bool operator<(const FractionalResourceQuantity &rhs) const;
+  bool operator>(const FractionalResourceQuantity &rhs) const;
+  bool operator<=(const FractionalResourceQuantity &rhs) const;
+  bool operator>=(const FractionalResourceQuantity &rhs) const;
+
+  /// \brief Return actual resource amount as a double.
+  double ToDouble() const;
+
+ private:
+  /// The resource name with the associated quantity. This is used only for
+  /// error messages.
+  std::string resource_name_;
+  /// The resource quantity represented as 1/10,000th of a unit. This value
+  /// will never exceed 10,000 because all resources are either in [0, 1] or
+  /// an integer.
+  int resource_quantity_;
+  /// Conversion factor that is the amount in internal units is equivalent to
+  /// one actual resource. Multiply to convert from actual to interal and
+  /// divide to convert from internal to actual.
+  static constexpr double conversion_factor_ = 10000;
```

Comment:
Let's put this in `src/ray/constants.h`.

Reply:
Or you can put it up above where `const std::string kCPU_ResourceLabel = "CPU";` is.